### PR TITLE
[http-client-csharp] Make RequestOptions parameter optional in protocol methods

### PR DESCRIPTION
**Pull request by jorgerangel-msft**
fixes: https://github.com/microsoft/typespec/issues/4990

contributes to https://github.com/microsoft/typespec/issues/4903

### CODE REVIEW

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs [hunk 1]
```diff
@@ -431,5 +453,31 @@ private static CSharpType GetConvenienceReturnType(IReadOnlyList<OperationRespon
...
+
+            for (int i = 0; i < convenienceMethodParameterCount; i++)
+            {
+                if (!MethodParameters[i].Type.Equals(ConvenienceMethodParameters[i].Type))
+                {
+                    return true;
+                }
+            }
+
+            return protocolReturnType?.Equals(convenienceReturnType) == true;
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs:480]: Why does this matter?

Comment (jorgerangel-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs:480]: I was just trying to match the [existing autorest behavior](https://github.com/Azure/autorest.csharp/blob/main/src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs#L207). If this isn't needed, or is a bug in autorest, I can remove.

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs [hunk 2]
```diff
@@ -431,5 +453,31 @@ private static CSharpType GetConvenienceReturnType(IReadOnlyList<OperationRespon
                 ? ClientModelPlugin.Instance.TypeFactory.ClientResponseApi.ClientResponseType
                 : new CSharpType(ClientModelPlugin.Instance.TypeFactory.ClientResponseApi.ClientResponseOfTType.FrameworkType, responseBodyType);
         }
+
+        private bool ShouldAddOptionalRequestOptionsParameter(CSharpType? protocolReturnType, CSharpType? convenienceReturnType)
+        {
+            var convenienceMethodParameterCount = ConvenienceMethodParameters.Count;
+            if (convenienceMethodParameterCount == 0 && protocolReturnType?.Equals(convenienceReturnType) == false)
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs:460]: Why does the return type matter here?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs [hunk 3]
```diff
@@ -673,6 +700,54 @@ public static IEnumerable<TestCaseData> BuildConstructorsTestCases
             }
         }
 
+        public static IEnumerable<TestCaseData> RequestOptionsParameterInSignatureTestCases
+        {
+            get
+            {
+                yield return new TestCaseData(
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs:707]: Can we add a test case where there are parameters but the RequestOptions should not be optional?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs (hunk 1 above)
Comment (JoshLove-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs:480]: I don't think this is relevant here.

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs (hunk 2 above)
Comment (jorgerangel-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs]: removed.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**